rosidl_typesupport_introspection_c/include/rosidl_typesupport_introspection_c/message_introspection.h

```diff
@@ -24,31 +24,65 @@
 
 #include "rosidl_typesupport_introspection_c/visibility_control.h"
 
+/// Structure used to describe a single field of an interface type.
 typedef struct rosidl_typesupport_introspection_c__MessageMember_s
 {
+  /// The name of the field.
   const char * name_;
+  /// The type of the field as a value of the field types enum
+  /// rosidl_typesupport_introspection_c_field_types.
+  /// e.g. rosidl_typesupport_introspection_c__ROS_TYPE_FLOAT
   uint8_t type_id_;
+  /// If the field is a string, the upper bound on the length of the string.
   size_t string_upper_bound_;
+  /// If the type_id_ value is rosidl_typesupport_introspection_c__ROS_TYPE_MESSAGE,
+  /// this points to an array describing the fields of the sub-interface.
   const rosidl_message_type_support_t * members_;
+  /// True if this field is an array type, false if it is any other type. An
+  /// array has the same value for / type_id_.
   bool is_array_;
+  /// If is_array_ is true, this contains the number of members in the array.
   size_t array_size_;
+  /// If is_array_ is true, this specifies if the array has a maximum size. If it is true, the
+  /// value in array_size_ is the maximum size.
   bool is_upper_bound_;
+  /// The bytes into the interface's in-memory representation that this field can be found at.
   uint32_t offset_;
+  /// If the interface has a default value, this points to it.
   const void * default_value_;
+  /// If is_array_ is true, a pointer to a function that gives the size of one member of the array.
+  /// Only used for nested interface types.
```

Comment:
It also seems to be used for normal types too. I see it set for a double array on the CameraInfo message. Same comment on the other fields below.

```C
  {
    "p",  // name
    rosidl_typesupport_introspection_c__ROS_TYPE_DOUBLE,  // type
    0,  // upper bound of string
    NULL,  // members of sub message
    true,  // is array
    12,  // array size
    false,  // is upper bound
    offsetof(sensor_msgs__msg__CameraInfo, p),  // bytes offset in struct
    NULL,  // default value
    sensor_msgs__msg__CameraInfo__rosidl_typesupport_introspection_c__size_function__CameraInfo__p,  // size() function pointer
    sensor_msgs__msg__CameraInfo__rosidl_typesupport_introspection_c__get_const_function__CameraInfo__p,  // get_const(index) function pointer
    sensor_msgs__msg__CameraInfo__rosidl_typesupport_introspection_c__get_function__CameraInfo__p,  // get(index) function pointer
    sensor_msgs__msg__CameraInfo__rosidl_typesupport_introspection_c__fetch_function__CameraInfo__p,  // fetch(index, &value) function pointer
    sensor_msgs__msg__CameraInfo__rosidl_typesupport_introspection_c__assign_function__CameraInfo__p,  // assign(index, value) function pointer
    NULL  // resize(index) function pointer
  },
```

Reply:
@j-rivero this one is still outstanding - I think github collapsed some of the review comments. There are a few more besides this one.

Reply:
7f1d585f3190735ba8d9cf943db57a8abb675737